Firmware/MotorControl/axis.cpp

```diff
@@ -424,13 +421,16 @@ bool Axis::run_homing() {
         return false;
     }
 
-    // Set the current position to 0.
-    encoder_.set_linear_count(0);
-    controller_.input_pos_ = 0;
+    // Set the current position to 0, the target to zero, and make sure we're path planning from 0 to 0
+    encoder_.set_linear_count(0); 
+    controller_.input_pos_ = 0.0f;
+    controller_.pos_setpoint_ = 0.0f;
+    controller_.vel_setpoint_ = 0.0f;
     controller_.input_pos_updated();
 
-    controller_.config_.control_mode = stored_control_mode;
-    controller_.config_.input_mode = stored_input_mode;
+    // Force encoder estimate to update
+    osDelay(1);
```

Comment:
We have `wait_for_control_iteration()` which would be fitting here

Reply:
Since this is legacy homing and it was tested with osDelay, I think I'd rather just keep it